interview-questions.md

```diff
@@ -8,44 +8,46 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What does calling super do in a class?
 
-Your answer:
+Your answer: Calls an attribute from the initialized class
 
-Researched answer:
+Researched answer: calling super will call the initialize method for that attr.
+my understanding is that the class has an initial set of attr that can be used with any instance of that class. so, calling super and that attr will simply give you that attr the way it was initialy initialized.
 
 2. What is a gem?
 
-Your answer:
+Your answer: something you can download that has functionality that other people created.
 
-Researched answer:
+Researched answer: packaged up ruby code into A library of code functionality managed by RubyGems.
+to make it easier to implement in an application and to share with other developers
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: a place where a lot of information is stored and can be retrieved from.
 
-Researched answer:
+Researched answer: a schema that is like a set of rows and columns with information also knows as tables that can have relatable info to other tables
 
 4. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: an attribute that classes have to distinguish them from other very identical instinces of the similar class.
 
-Researched answer:
+Researched answer: the row of a table's id that is unique and not shared with any other table.
```

Comment:
A row would usually represent an instance, where as the primary key would be a column providing a unique identifier for each instance in the database.